src/compiler/checker.ts

```diff
@@ -23132,9 +23132,14 @@ namespace ts {
 
         function isConstantReference(node: Node): boolean {
             switch (node.kind) {
-                case SyntaxKind.Identifier:
+                case SyntaxKind.Identifier: {
                     const symbol = getResolvedSymbol(node as Identifier);
-                    return isConstVariable(symbol) || !!symbol.valueDeclaration && getRootDeclaration(symbol.valueDeclaration).kind === SyntaxKind.Parameter && !isParameterAssigned(symbol);
+                    if (isConstVariable(symbol)) {
+                        return true;
+                    }
+                    const decl = symbol.valueDeclaration ? getRootDeclaration(symbol.valueDeclaration) : undefined;
+                    return !!decl && (isParameter(decl) || isCatchClauseVariableDeclaration(decl)) && !isSymbolAssigned(symbol);

```

Comment:
This predicate (except `!isSymbolAssigned`) is now duplicated. Do you think it's worth extracting to a function?

Reply:
Good point. Thanks. I've moved it to function.